### PR TITLE
Add OpenMapSamples

### LINKED ISSUE #135: Add an OpenMapSamples control to the demo
I’m happy to report that I’ve now built a prototype MapLibre control that can temporarily replace the data sources used by a map style with dynamically generated GeoJSON:
* https://github.com/adamfranco/OpenMapSamples-MapLibre
* https://github.com/adamfranco/OpenMapSamples

A demo is available at https://adamfranco.github.io/OpenMapSamples-MapLibre/

The reason I wanted to dynamically generate the data rather than hand-coding it can be seen in the definition of the [HighwayAttributes sample](https://github.com/adamfranco/OpenMapSamples/blob/main/samples/OpenMapTiles/HighwayAttributes.js). A simple two-dimensional array of tag combinations is all that’s needed to generate sample data that is spaced out to be visible at 16 different zoom levels. To hand-map this one sample’s examples would have required mapping 2016 individual ways. :wink:

The next step is to iron out some more example data to add labels and improve coverage. The [OpenMapTiles schema documentation](https://openmaptiles.org/schema/) isn’t quite sufficient to figure out what is needed for a more realistic mock — it doesn’t note data types or full property formatting and seems a bit stale as well. 

Once this control and the samples are more fleshed out we can wire up this control in the Americana demo so that we can jump around to different examples as we work on style definition.

One thing I ran into in a first attempt at including this control in Americana is that the current `make` build process doesn't do any bundling step, so references to dependent `node_modules/` don't get resolved. #134 will help with that, but we may need to bundle the demo with either [Parcel](https://parceljs.org/) (what I've used for my projects), or [WebPack](https://webpack.js.org/). See https://levelup.gitconnected.com/parcel-vs-webpack-2021-64c347bcb31 for a discussion of the pros & cons of each.

### PR DESCRIPTION
**Pull request by adamfranco**
This is a PR to add the [OpenMapSamples-Maplibre](https://github.com/adamfranco/OpenMapSamples-MapLibre) control to the Americana demo. Will fix #135.

Still to do:
- [x] Complete and merge #137 so that this control and its dependencies can be included.
- [x] Shift the samples button to not conflict with the MapTiler attribution.

### CODE REVIEW

**Comment by adamfranco** [comment]
More samples are now ready for use. 🎉 

Screen Shot 2022-02-08 at 6 43 57 PM
Screen Shot 2022-02-08 at 6 42 06 PM
Screen Shot 2022-02-08 at 6 41 50 PM
Screen Shot 2022-02-08 at 6 41 35 PM

**Comment by adamfranco** [comment]
Ok, this PR is now installing reliably now that I've figured out how to force the `openmapsamples-maplibre` dependency to use our forked `maplibre-gl` instead of any other version. This require npm 8.30+ to get the new `overrides` feature, but that is the current stable version, so that shouldn't be an issue.

File: style/package.json [hunk 1]
```diff
@@ -10,16 +10,25 @@
...
   },
   "devDependencies": {
-    "browser-sync": "^2.27.7",
+    "maplibre-gl": "https://github.com/ZeLonewolf/maplibre-gl-js/releases/download/v2.0.5/maplibre-gl.tar.gz",
+    "openmapsamples-maplibre": "github:adamfranco/OpenMapSamples-MapLibre",
+    "parcel": "^2.3.0",
     "prettier": "2.3.2"
+  },
+  "engines": {
+    "npm": ">=8.3.0"
```
Comment (1ec5) [inline style/package.json:29]: Can you add an .nvmrc file containing something like `stable` for those who use `nvm` to manage their Node versions? Looks like `engines` doesn’t enforce the version that `nvm` uses, unfortunately: nvm-sh/nvm#651.

Comment (adamfranco) [reply style/package.json:29]: We could do that, but it seems that `stable` is a deprecated alias: https://stackoverflow.com/questions/68357267/nvm-use-stable-sets-node-version-to-older-than-version-installed-by-nvm-insta

We also probably don't want to hard-code to a single Node version, just provide a minimum version, but I'm open to suggestions. :-)

**Comment by adamfranco** [comment]
I broke the build and deploy action, but #152 should fix it. Once that is in place this should merge and deploy cleanly. 🤞

**Comment by ZeLonewolf** [comment]
An `npm start` failed out of the box for me:

image

**Comment by adamfranco** [comment]
Did you run `npm i --include=dev` again after checking out the branch?

**Comment by ZeLonewolf** [comment]
That gives me:
```
brian@laptop:~/git/openstreetmap-americana/style$ npm i --include=dev
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for openstreetmap-americana@0.1.0: wanted: {"npm":">=8.3.0"} (current: {"node":"14.18.2","npm":"6.14.15"})
npm ERR! notsup Not compatible with your version of node/npm: openstreetmap-americana@0.1.0
npm ERR! notsup Not compatible with your version of node/npm: openstreetmap-americana@0.1.0
npm ERR! notsup Required: {"npm":">=8.3.0"}
npm ERR! notsup Actual:   {"npm":"6.14.15","node":"14.18.2"}

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/brian/.npm/_logs/2022-02-11T14_50_05_834Z-debug.log
```

**Comment by ZeLonewolf** [comment]
Ideally we should be able to hide this low-level setup stuff behind top-level run commands.

**Comment by ZeLonewolf** [comment]
Okay, apparenly I'm getting this now regardless of branch.  Really annoying how npm seems to always end up in some kind of version hell that you can't escape from.

**Comment by adamfranco** [comment]
Are you using nvm to install node/npm or a global install of node?

https://github.com/nvm-sh/nvm

**Comment by adamfranco** [comment]
I hadn't heard of nvm until fighting with getting the right versions set up
on Apple m1. It lets you install many versions of node and switch which one
is active. We should probably update the install instructions.

File: style/package.json (hunk 1 above)
Comment (zekefarwell) [reply style/package.json:29]: 8.3 here is also the [NPM version](https://github.com/npm/cli/releases) not the [Node version](https://nodejs.org/en/download/releases/).   I don't think NVM helps with manageing NPM versions, but I could be wrong. NPM can update itself to the latest version with `npm update -g npm`.

**Comment by zekefarwell** [comment]
> We should probably update the install instructions.

I can do this later today.  I like writing documentation.

**Review by zekefarwell — APPROVED** [review]
Seems to work fine for me.  I tested with NPM 8.4 and 8.5 running on Node 14 and 16.

**Pull request merged**